src/Package/Impl/TaskList/VsTaskItem.cs

```diff
@@ -53,8 +54,8 @@ internal sealed class VsTaskItem : ErrorTask {
                     break;
             }
 
-            IVsHierarchy hierarchy = _source.TextBuffer.GetHierarchy();
-            base.HierarchyItem = hierarchy;
+            var textBuffer = _source.EditorBuffer as ITextBuffer;
```

Comment:
This will turn into 'IEditorBuffer` soon rather than the object so doesn't matter much at the moment. However, it should do null checks since task list source is not necessarily an editor document.